Sitecore/jss
# Feature/467450 2

Update GraphQL API per latest edge schema
- All queries need to receive a language (updated connected/integrated GQL components)
- IDs passed passed to queries no longer need to be lowercased, all formats are now accepted
- No longer need to specify sort order for search, as there is a default sort order on the backend

Improvements
- `getAppRootId` helper is now used by both, sitemap service and gql dictionary service
- Functionality for executing search query that was repeated in multiple places (pagination, validating params) has been pulled out into `SearchService`, which can be used by other services
- New interface for GraphQLClient so that services can chose to use a different graphQL library if they want (no longer coupled to graphql-request)
- Apply similar patter to CacheClient: enable services to chose a different cache library
- Make it easier to customize our services by changing access level of private methods to protected

Testing Details
Unit tests updated for API changes, and more tests added

Documentation
Docs for new services coming in another PR
